syntax = "proto3";

package bindle;

service Bindle {

	rpc GetValue(Key) returns (GetResult) {}

	rpc InsertValue(KeyValue) returns (InsertResult) {}

	rpc RemoveKey(Key) returns (RemoveResult) {}

}

message Key {
	string key = 1;
}

message KeyValue {
	string key = 1;
	string value = 2;
}

message GetResult {
	optional string value = 1;
}

message InsertResult {}

message RemoveResult {}
